git/url.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2024 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package git
+
+import (
+	"net/url"
+	"os"
+	"strings"
+	"unicode"
+
+	"github.com/terramate-io/terramate/errors"
+)
+
+// ErrInvalidGitURL indicates is an error kind indicating the git URL is not valid.
+const ErrInvalidGitURL errors.Kind = "invalid git remote URL"
+
+// Repository aggregates information about a repository.
+type Repository struct {
+	RemoteURL string // original remote URL (eg.: git@github.com:terramate-io/terramate.git)
```

Comment:
Could it be the name misleading as it may be not remote?
```suggestion
	RawURL string // original raw URL (eg.: git@github.com:terramate-io/terramate.git)
```

Reply:
Git calls it [remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes). For example: `git remote get-url origin` can return a URL or a path like `file://<path>`.
But `RawURL` is also a good name, which also indicates it holds whatever comes back from the `git remote get-url` command (unmodified). I will change. Thank you.